library/src/main/kotlin/com/connectrpc/protocols/GRPCInterceptor.kt

```diff
@@ -142,7 +143,7 @@ internal class GRPCInterceptor(
             streamResultFunction = { res ->
                 val streamResult = res.fold(
                     onHeaders = { result ->
-                        val responseHeaders = result.headers.filter { entry -> !entry.key.startsWith("trailer") }.toMutableMap()
+                        val responseHeaders = result.headers.filter { entry -> !entry.key.startsWith("trailer") }
```

Comment:
I think the `onCompletion` parsing is missing the same headers handling as unary now has?

Reply:
Ideally this would surface with a failing conformance test. Do you know if there is one that we're perhaps missing here?

Reply:
I believe `failStreamingOutputCall` and `failUnaryCall` fail, but only when the server is `grpc-go`, they pass with `connect-go` since `connect-go` uses trailers

Reply:
Added a test which failed because this was missing for streaming calls, and fixed the issue in the streaming implementation.